Fix the namespaced controller_manager spawner + added tests

saikishor opened the pull request:
This PR aims to fix the spawner to work with namespace controller manager and added tests for this particular case. It is started from #1547, however after looking into the spawner, it seems like the `--namespace` parameter is no longer required as you could pass it in the arg `--controller-manager` as how it is done for the unspawner, if this is fine, then we can get rid of the `--namespace` arg 

closes: #1547
fixes: #1506

christophfroehlich reviewed (changes requested):
Tested with https://github.com/ros-controls/ros2_control_demos/pull/502 but I still failed.

bmagyar reviewed (changes requested):
Sounds like a good idea but let's deprecate the namespace parameter first, loudly

saikishor commented:
> Sounds like a good idea but let's deprecate the namespace parameter first, loudly

Make sense. You are right. I shouldn't have removed it abruptly hahHa 😅😅

saikishor commented:
@bmagyar I've added the `--namespace` arg is now added as deprecated and I've also added tests for the deprecated arg

@christophfroehlich I've made some changes on https://github.com/ros-controls/ros2_control_demos/pull/502 for the new namespacing and also the missing `--param-file` arg. Check and let me know

christophfroehlich reviewed (changes requested):
Tested now successfully with [1ac5e13](https://github.com/ros-controls/ros2_control_demos/pull/502/commits/1ac5e139a371d1037c974698599ab653b6f87ef9) of https://github.com/ros-controls/ros2_control_demos/pull/502.

May I ask for updating the [docs](https://github.com/ros-controls/ros2_control/blob/master/controller_manager/doc/userdoc.rst#helper-scripts), and adding migration notes?

saikishor commented:
> May I ask for updating the [docs](https://github.com/ros-controls/ros2_control/blob/master/controller_manager/doc/userdoc.rst#helper-scripts), and adding migration notes?

Sire, I can update the docs.

christophfroehlich reviewed (dismissed):
Thanks :)

bmagyar commented on controller_manager/doc/userdoc.rst:
```diff
@@ -158,14 +158,16 @@ There are two scripts to interact with controller manager from launch files:
                             Controller param file to be loaded into controller node before configure
       -n NAMESPACE, --namespace NAMESPACE
-                            Namespace for the controller
+                            Namespace for the controller_manager and the controller(s) (deprecated)
```
```suggestion
                            DEPRECATED Namespace for the controller_manager and the controller(s)
```

bmagyar commented on controller_manager/controller_manager/spawner.py:
```diff
@@ -170,7 +173,11 @@ def main(args=None):
-        "-n", "--namespace", help="Namespace for the controller", default="", required=False
+        "-n",
+        "--namespace",
+        help="Namespace for the controller_manager and the controller(s) (deprecated)",
```
```suggestion
        help="DEPRECATED Namespace for the controller_manager and the controller(s)",
```
